scripts/0-preinstall.sh

```diff
@@ -69,19 +69,43 @@ echo -ne "
                     Creating Filesystems
 -------------------------------------------------------------------------
 "
+createswapfile() {
+    $SWAPFILE || return 0
+    if [[ "${FS}" == "ext4" ]]; then
+        local path=/opt/swap
+        mkdir -p /mnt$path # Put swap into the actual system, not into RAM disk, otherwise there is no point in it, it'll cache RAM into RAM. So, /mnt/ everything.
+        chattr +C /mnt$path #apply NOCOW, btrfs needs that.
+    else
+        local path=/swap
+        truncate -s 0 /mnt$path/swapfile
+        chattr +C /mnt$path/swapfile #apply NOCOW, btrfs needs that.
+        btrfs property set /mnt$path/swapfile compression none
+    fi  
+    dd if=/dev/zero of=/mnt$path/swapfile bs=1M count=4096 status=progress
+    chmod 600 /mnt$path/swapfile # set permissions.
+    chown root /mnt$path/swapfile
+    mkswap /mnt$path/swapfile
+    swapon /mnt$path/swapfile
+    # The line below is written to /mnt/ but doesn't contain /mnt/, since it's just / for the system itself.
+    echo "$path/swapfile	none	swap	sw	0	0" >> /mnt/etc/fstab # Add swap to fstab, so it KEEPS working after installation.
+    echo "vm.swappiness=10" >> /mnt/etc/sysctl.conf # Lower swappiness
+}
+
 createsubvolumes () {
     btrfs subvolume create /mnt/@
     btrfs subvolume create /mnt/@home
     btrfs subvolume create /mnt/@var
     btrfs subvolume create /mnt/@tmp
     btrfs subvolume create /mnt/@.snapshots
+    $SWAPFILE && btrfs subvolume create /mnt/@swap
```

Comment:
getting value of true or false and then executing the command is neat. But may not produce the same results. while using if statement is more likely to produce desired results.